### PR TITLE
Custom style from shield.io and some refactoring

### PR DESCRIPTION
**Pull request by uniibu**
Status: 
- Travis-ci Build passed Travis-ci

Major Changes:
- Added support to shields.io's style parameters and this should also satisfy https://github.com/greenkeeperio/greenkeeper/issues/416. With this PR, greenkeeper can now accept a `style` paramater and supports `flat` `flat-square` and `plastic` style from shields.io.
- Change the word `enabled` to `monitored`

Minor Changes:
- Refactored and rearranged some codes
- Changed 'Greenkeeper' word on the badge to the standard badge lettering  which is all lower-case.
- Added example badges on README
- Change left-side badge color to make the badge unique
- Added tests for the new style query
- Update Hapi.js version on the package.json
- Added package-lock.json on gitignore.
- Since the engine recommends the Node version to be 7, i would assume, its the latest of the v 7 releases, and version lower than 7.10.1 are labeled to be `high severity vulnerability` [july-2017-security-releases](https://nodejs.org/en/blog/vulnerability/july-2017-security-releases/#_-update-11-july-2017-_-security-releases-available), i've set the recommended version to `^7.10.1 || >= 8.1.4`
- And because of the node version change, `--harmony` flag is no longer needed, so i removed it.

Badges will look like the following:(flat-square,flat, plastic)

Greenkeeper flat Greenkeeper NotFound  Greenkeeper Disabled 

PS. Feel free to edit ☺️

### CODE REVIEW

**Comment by levithomason** [comment]
It would be great to get statuses similar to what gemnasium supports like "up to date" and "out of date".

May also consider sticking with the usual dark gray background on the left:



>https://github.com/Semantic-Org/Semantic-UI-React

Others looking may consider using something like https://david-dm.org instead.

**Comment by j-f1** [comment]
Why not use the Greenkeeper green for the enabled/monitored background color? It’s already green, so it’d be less of a change. You could also change the left side to say “🌴 greenkeeper” to stay closer to the brand.

**Comment by kbrandwijk** [comment]
I am a big fan of the 'for-the-badge' style. Any reason why that's not included?

File: lib/badges.js [hunk 1]
```diff
@@ -1,127 +1,103 @@
 const crypto = require('crypto')
-
```
Comment (janl) [inline lib/badges.js:2]: Please do whitespace-only changes in a separate commit.

This allows us to skim over whitespace-only changes in larger functions (see below)

File: lib/badges.js [hunk 2]
```diff
@@ -1,127 +1,103 @@
...
 const _ = require('lodash')
-
```
Comment (janl) [inline lib/badges.js:4]: Whitespace

File: lib/badges.js [hunk 3]
```diff
@@ -1,127 +1,103 @@
...
-  const disabled = await ghBadges({
-    format: 'svg',
-    template: 'flat',
-    colorscheme: 'lightgray',
-    text: [
-      'Greenkeeper',
-      'disabled'
-    ]
+// error second API like a pro
+const ghBadges = data =>
```
Comment (janl) [inline lib/badges.js:8]: This is one of the candidates for why we prefer whitespace-only changes that don’t commit any functional changes:

 
screen shot 2018-03-06 at 12 04 19

File: lib/badges.js [hunk 4]
```diff
@@ -1,127 +1,103 @@
...
+    try {
+      ghBadgesCb(data, (result, err) => {
+        if (err) return reject(err)
+        resolve(result)
+      })
+    } catch (err) {
+      reject(err)
+    }
   })
+const createHash = bytes => crypto.createHash('sha256').update(bytes).digest('hex')
```
Comment (janl) [inline lib/badges.js:19]: This is a pure style change that makes the intent hard to follow as well.

File: lib/badges.js [hunk 5]
```diff
@@ -1,127 +1,103 @@
...
+    paymentRequired: getAllEtags(paymentRequired)
   }
 
-  function getBadge (repoDoc, payDoc) {
-    if (!repoDoc.enabled) return [disabled, etags['disabled']]
-    if (!repoDoc.private) return [enabled, etags['enabled']]
-    if (!payDoc) return [paymentRequired, etags['paymentRequired']]
-    if (payDoc.plan === 'org' || payDoc.plan === 'personal') return [enabled, etags['enabled']]
-    return [paymentRequired, etags['paymentRequired']]
+  const getBadge = (repoDoc, payDoc, style) => {
```
Comment (janl) [inline lib/badges.js:54]: Whitespace & Style-changes mixed with functional changes again.

File: lib/badges.js [hunk 6]
```diff
@@ -1,127 +1,103 @@
...
-    handler: handleBadges
-  })
-
-  async function handleBadges (request, reply) {
-    const {repositories, payments} = await dbs()
-
+  const handleBadges = async (request, reply) => {
+    const { repositories, payments } = await dbs()
     const key = request.paramsArray.join('/').toLowerCase()
+    let repoDoc, payDoc
```
Comment (janl) [inline lib/badges.js:68]: Whitespace again.

File: package.json [hunk 7]
```diff
@@ -9,7 +9,7 @@
     "good": "^7.0.2",
     "good-console": "^6.3.1",
     "good-squeeze": "^5.0.0",
-    "hapi": "^16.1.0",
```
Comment (janl) [inline package.json:12]: Please add a reason for updating the hapi version? Are you using hapi features that are not available earlier?

Also, this should be a separate commit.

File: package.json [hunk 8]
```diff
@@ -22,7 +22,7 @@
     "tap": "^10.0.1"
   },
   "engines": {
-    "node": "7"
```
Comment (janl) [inline package.json:25]: Please add a reason for updating the engine version? Are you using features that are not available earlier?

Also, this should be a separate commit.

File: package.json [hunk 9]
```diff
@@ -36,8 +36,8 @@
     "db:start": "./start-couchdb",
     "deploy": "./deploy",
     "pretest": "standard && npm run db:start && sleep 5",
-    "start": "node --harmony_async_await index.js",
```
Comment (janl) [inline package.json:39]: This goes with the engine-changes, should be separate from the main commit here.

File: test/badges.js [hunk 10]
```diff
@@ -4,32 +4,33 @@ const hapi = require('hapi')
 const tap = require('tap')
 
 const register = require('../lib/badges')
-const dbs = require('../lib/dbs')
```
Comment (janl) [inline test/badges.js:7]: Whitespace again

File: .gitignore [hunk 11]
```diff
@@ -1,3 +1,4 @@
 node_modules/
 .env
 .nyc_output/
+package-lock.json
```
Comment (janl) [inline .gitignore:4]: Why do you remove the package-lock.json?

File: .travis.yml [hunk 12]
```diff
@@ -8,6 +8,7 @@ cache:
   - node_modules
 node_js:
   - '7'
+  - '8'
```
Comment (janl) [inline .travis.yml:11]: Greenkeeper ops is not running on node 8 just yet, so we’d rather keep an engine update out of this PR until we move over the whole project.

File: package.json [hunk 13]
```diff
@@ -22,7 +22,7 @@
     "tap": "^10.0.1"
   },
   "engines": {
-    "node": ">=7.6"
+    "node": "^7.10.1 || >= 8.1.4"
```
Comment (janl) [inline package.json]: Can you explain why these versions are picked exactly?

**Review by janl — CHANGES_REQUESTED** [review]
Heya, thanks for taking the time to improve Greenkeeper, and apologies for the late response here.

This is a good addition overall, we just request a few style changes. I’ve added inline comments for the few details. If you could help clean up the commits a little bit, I’m sure we can get this merged quickly.

**Comment by uniibu** [comment]
This pull request was from an older version of greenkeeperio/badges (Aug. 2017). Im closing this this pull request and creating a new one to keep things clean. I've also noted your comments @janl and will update accordingly. Cheers

**Pull request closed without merging**